benchexec/tools/ultimate.py

```diff
@@ -105,24 +108,21 @@ def _ultimate_version(self, executable):
                 "--version",
             ],
             # 1
-            [
-                self.get_java(),
-                "-Xss4m",
-                "-jar",
-                launcher_jar,
-                "-data",
-                data_dir,
-                "--version",
-            ],
+            ["-Xss4m", "-jar", launcher_jar, "-data", data_dir, "--version"],
         ]
 
         self.api = len(cmds)
         for cmd in cmds:
-            version = self._query_ultimate_version(cmd, self.api)
-            if version != "":
-                return version
+            for java_version, java in java_versions.items():
+                version = self._query_ultimate_version([java] + cmd, self.api)
+                if version != "":
+                    logging.debug(
+                        f"Using Java {java} for API version {self.api} of Ultimate {version}"
```

Comment:
f-strings don't work on Python 3.5, which is still supported. Could you please avoid them? The logging module has lazy builtin string formatting anyway.
```suggestion
                       "Using Java %s for API version %s of Ultimate %s", java, self.api, version
```
(Also for the other log statements, of course.)